Composer/packages/adaptive-flow/src/adaptive-flow-editor/components/ZoomZone.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import { useRef, useEffect, ReactNode } from 'react';
+import { ZoomInfo } from '@bfc/shared';
+import { IconButton, IButtonStyles } from 'office-ui-fabric-react/lib/Button';
+import { IIconProps } from 'office-ui-fabric-react/lib/Icon';
+
+function ScrollZoom(
```

Comment:
the function name needs to be camel-cased

Reply:
And we can make this function pure by not operating DOM inside it. The DOM operation can be hoist to `handleZoome` which can consume component context.

Reply:
nit: another suggestion is, we can wrap the zoom rate calculation logic as a recoil action.
where it exposes two interfaces `zoomIn()` and `zoomOut`, they manipulates the recoil value directly.

Reply:
I think that the zoom logics are wraped in the zoomZone is more flexable because we need use the newest rate to excute zoom in/out for the flow.  Recoil action just set recoil state.